### ISSUE TITLE
Retreating operators in Trials for Navigator not working properly with Copilot

### ISSUE THREAD

**Issue by ajk992** [MaaAssistantArknights/MaaAssistantArknights#10282]
### Before raising the issue...

- [X] I understand that Issues are for feedback and problem solving, not trolling the comments section, and will provide as much information as possible to help solve the problem.
- [X] I filled in a short, clear title so that developers could quickly identify the general problem when going through the issue list. Instead of "Some suggestions", "Stuck", etc.
- [X] I have checked the latest beta update and there is no mention of the bug being fixed.
- [X] I have reviewed the pinned issues and confirmed that my issue is not mentioned.

### Description

When running copilot JSON files for TN-1, TN-2, TN-3, or TN-4 stages, operators aren't being properly retreated. It seems like they're being tapped on and then nothing happens afterwards? This doesn't seem to be an issue with normal story/event stages. I know TN is a temporary event, but I figured I'd report it.

Copilot JSON used in video: maa://33584

### Version

Resource Version: Trials for Navigator
Core Version: v5.5.11452

### Log files

[gui.log](https://github.com/user-attachments/files/16651484/gui.log)
[asst.log](https://github.com/user-attachments/files/16651485/asst.log)

### Configuration information

[gui.json](https://github.com/user-attachments/files/16651487/gui.json)
Windows 10
Bluestacks 5
1920x1080
240 DPI
60 FPS

### Screenshots

https://github.com/user-attachments/assets/f3d4886c-6f7a-400c-a726-15c17fa27dc3
Video example, sorry if the quality isn't good. Typhon should've been retreated after 10 kills, but it didn't work properly.

**Comment by Constrat** [comment]
Yeah experiencing the same issue. At first I was thinking it was a lag issue as the stages are a bit more laggy than usual, but after trying directly on native Android the same remains, it appears that the click happens outside the retreat button.
I've mentioned this on telegram while I was investigating privately, but in the end I arrived to no solution:

I was thinking that the retreat button position calculation might be off here:
```cpp
    m_map_data = TilePack::find_level(stage_name).value_or(Map::Level {});
    auto calc_result = TilePack::calc(stage_name, shift_x, shift_y);
    m_normal_tile_info = std::move(calc_result.normal_tile_info);
    m_side_tile_info = std::move(calc_result.side_tile_info);
    m_retreat_button_pos = calc_result.retreat_button;
    m_skill_button_pos = calc_result.skill_button;
```

**Comment by Constrat** [comment]
For once I don't think this is only EN related.
@HX3N @Manicsteiner if you are able to check (It would be crazy if it was only YostarEN, the usual suspect)

**Comment by HX3N** [comment]
> For once I don't think this is only EN related. @HX3N @Manicsteiner if you are able to check (It would be crazy if it was only YostarEN, the usual suspect)

Same issues in KR

**Comment by Constrat** [comment]
cc @status102 I remember you had to modify the TN stage values. Could this be related?

**Comment by status102** [comment]
> cc @status102 I remember you had to modify the TN stage values. Could this be related?

I'm not sure which stage is TN? could you give me a screenshot

**Comment by Constrat** [comment]
The trials for navigator, the boss rush with the 3 different bosses where the map slides. The issue author posted a video of it (TN-2)

**Comment by status102** [comment]
oh, I saw it. I'm not sure if yj have changed it.
You may verify views value with https://github.com/yuanyan3060/Arknights-Tile-Pos

**Comment by Constrat** [comment]
They are indeed different. 
![image](https://github.com/user-attachments/assets/35bc099f-8ffb-4146-ad9a-6cdb98f52665)
But... @status102 you made them different with 647cf6d656f50f2206f11f0d6bb6ab6fda6e05f7

**Comment by status102** [comment]
> They are indeed different. ![image](https://private-user-images.githubusercontent.com/56174894/359100984-35bc099f-8ffb-4146-ad9a-6cdb98f52665.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNzIzOTMsIm5iZiI6MTcyNDA3MjA5MywicGF0aCI6Ii81NjE3NDg5NC8zNTkxMDA5ODQtMzViYzA5OWYtOGZmYi00MTQ2LWFkOWEtNmNkYjk4ZjUyNjY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDEyNTQ1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1M2JjMmQ0MTMyNGZmMGZhNTZmZDA0YTUyYmM3MTQzMjNmMTMxNDc3N2M3NDQ5ODA3Nzc3MjBmNTg1YWU3NzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oTwBLyBxQX-IJmkvxufpR3KpQ57-kfPetpbDvMqPZB8) But... @status102 you made them different with [647cf6d](https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/647cf6d656f50f2206f11f0d6bb6ab6fda6e05f7). That commit introduced -3.56 instead of -3.48.
> 
> I think the tile pos was always -3.48 and you just changed them.
> 
> EDIT: I see, the resource updater didn't add anything and [4067c9d](https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/4067c9d369011f24660edf17a313c1e00f23a237) added the missing information, but apparently the missing information was "wrong"?
> 
> Anyway the same issue is happening with TN-1, and that wasn't modified in your commit so I don't think this is the issue. Yesterday I tried reverting your commit but it wasn't retreating anyway.

Anyway, all values in my commits are generated from manual gathering data. So there may be some deviations. Change it if u like

**Comment by Constrat** [comment]
If I change, I would be reverting to the values before your modification.

**Comment by status102** [comment]
Make a screenshot during deploying. Then, get pos of tiles. Finally, use pos to calculate views data.

In that repo, there is a process shows how to calculate pos using `views`. So I make it with a reverted process

**Comment by Constrat** [comment]
So your values are probably more accurante in a MAA related environment for the CN server... The only issue that remains is if the CN and Global have the same view zoom level then :P

**Comment by status102** [comment]
However, TN is not opened in CN now. Maybe change it and revert later is a better way

**Comment by Constrat** [comment]
Honestly I don't reverting is the solution. For now I'll leave everything as-is. I'll investigate if "new" values are required for global.

**Comment by Constrat** [comment]
Confirmed that the changes introduced with cd347b7 broke the Trials for Navigator copilot tasks.
Calculating the retreat button and skill position work great in all changes except TN ones, that being very wide and with moving cameras probably are not managed correctly.

Running MAA detached at 8c2d73d7 makes the tasks work correctly.
Unfortunately reverting the commit itself is impossible as a lot of files and commits depend on these changes.

This was never flagged before because after cd347b7 no Trials for Navigator have been released on the CN server, meaning the user base still hasn't discovered this issue. Trials for Navigator#5 will most likely not work.

**Comment by Constrat** [comment]
Attempting to revert starting from commit f5dea88 proved unsuccessful; the changes are too intertwined.
